### PR TITLE
General code cleanup part #1

### PR DESCRIPTION
**Pull request by parente**
* Adding docstrings everywhere
* Simplifying state tracking in a few place
* Removing dead code
* Addressing minor output problems

Prelude to #28, 26, and #22 which will build atop this but that
I want to keep separate for easier review.

### CODE REVIEW

**Comment by parente** [comment]
Whoops. Gotta put codecov back in.

File: spylon_kernel/init_spark_magic.py [hunk 1]
```diff
@@ -21,33 +32,42 @@ def __init__(self, kernel):
...
+        in the variable `launcher`. Looks for an `application_name`
+        variable to use as the name of the Spark session.
 
-        This will evaluate the launcher args using spylon.
+        Example
+        -------
+        %%init_spark
+        application_name = "My Fancy App"
+        launcher.jars = ["file://some/jar.jar"]
+        launcher.master = "local[4]"
```
Comment (ericdill) [inline spylon_kernel/init_spark_magic.py]: What in the world is this `local[4]` syntax?

Comment (parente) [inline spylon_kernel/init_spark_magic.py]: Spark syntax for how many cores to use. `local[4]` says "Please use 4 of my cores for the application master." `local[*]` says "TAKE ALL MY CORES PLZ!"

Comment (ericdill) [inline spylon_kernel/init_spark_magic.py]: Can you add that as a comment in the code?

**Comment by parente** [comment]
Going to merge what I've got, then carry on with documenting the other two modules in another PR. Thanks folks.

**Pull request merged**